src/Translator.php

```diff
@@ -144,6 +144,18 @@ private function setModulePaths(): void
                     continue;
                 }
                 $branch = $this->exec('git rev-parse --abbrev-ref HEAD', $modulePath);
+                // the following happens when a tag is checked out, rather than a minor or next minor branch
+                if ($branch === 'HEAD') {
+                    $cmd = "git branch 2> /dev/null | sed -e '/^[^*]/d' -e 's/* \(.*\)/\\1/'";
+                    // $detachedBranch example: "(HEAD detached at 3.0.2)"
+                    $detachedBranch = $this->exec($cmd, $modulePath);
+                    // try to git checkout the what the minor branch for the tag should be
+                    if (preg_match('#^\(HEAD detached at ([0-9]+\.[0-9]+)\.[0-9]+\)$#', $detachedBranch, $matches)) {
```

Comment:
We should have an else clause here that at the very least is a warning to the user and then skip this module, if not outright bail here.

For example if I checkout a commit hash, we'll still get to this point but we'll have something like `"(HEAD detached at b10762240)"` so this condition will be false, and we'll continue on down as though we had a branch.

That's fine with the current implementation because we'd hit the `!is_numeric()` failure but if we change the logic later we might forget about this scenario and it could bite us in the butt later on. Better to be explicit about this scenario, I think.

Reply:
If we bail here, we're just duplicating logic from the is_numeric() check directly below, which covers both the 'current branch is HEAD' and current branch is 'dev-pulls/5.0/some-bugfix' scenarios

Reply:
> That's fine with the current implementation because we'd hit the !is_numeric() failure but if we change the logic later we might forget about this scenario and it could bite us in the butt later on. Better to be explicit about this scenario, I think.

My comment above covers that objection

Reply:
I rewrote the block and got rid of the use of sed